src/riak_object.erl

```diff
@@ -129,6 +130,19 @@ new(B, K, V, MD) when is_binary(B), is_binary(K) ->
 newts(B, K, V, MD) ->
     new_int2(B, K, V, MD).
 
+-spec is_ts(riak_object()) -> {'true', pos_integer()} | 'false'.
+is_ts(RObj) ->
+    check_for_ddl(get_contents(RObj)).
+
+check_for_ddl([{Metadata, _V}]) ->
```

Comment:
Held a brief discussion with Fred, we’re going to do more digging but it does not appear safe to assume they never have siblings.

-John

> On Apr 26, 2016, at 11:09 AM, Andy Till <notifications@github.com> wrote:
> 
> In src/riak_object.erl <https://github.com/basho/riak_kv/pull/1381#discussion_r61103586>:
> 
> > @@ -129,6 +130,19 @@ new(B, K, V, MD) when is_binary(B), is_binary(K) ->
> >  newts(B, K, V, MD) ->
> >      new_int2(B, K, V, MD).
> >  
> > +-spec is_ts(riak_object()) -> {'true', pos_integer()} | 'false'.
> > +is_ts(RObj) ->
> > +    check_for_ddl(get_contents(RObj)).
> > +
> > +check_for_ddl([{Metadata, _V}]) ->
> Write Once objects never have siblings. Maybe better to crash than to silently ignore that, and lose data presumably if it ever happens.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/basho/riak_kv/pull/1381/files/ccfe689af543527b6c45d025ae07be13969ce672#r61103586>